x/wasm/types/context.go

```diff
@@ -14,8 +14,15 @@ const (
 	contextKeyQueryStackSize contextKey = iota
 	// authorization policy for sub-messages
 	contextKeySubMsgAuthzPolicy = iota
+	// gas register
+	contextKeyGasRegister = iota
 )
 
+type gasRegister interface {
```

Comment:
There is an interface `GasRegister` in the keeper package. Let's move this to `types` package and use it here. Otherwise it can get confusing to have both.